cmd/soroban-cli/src/strval.rs

```diff
@@ -15,7 +16,7 @@ use crate::utils;
 #[derive(Debug)]
 pub enum StrValError {
     UnknownError,
-    InvalidValue,
+    InvalidValue(Option<ScSpecType>),
```

Comment:
💭 An optional is fine for now, but I suspect we can probably replace any of the `None` cases because in every case we expect to know what the `ScSpecTypeDef` is. This doesn't need to block this PR, and is something we can follow up with.

Reply:
The thinking behind `None` is simply because as of today there are some cases where `I don't know` what the best error output (info) would be. The more I get to know Soroban and its type def, I could certainly add the correct error handling to these cases, therefore resulting in `None`, and `Option<>` being replaced for something better :)   -- such as having a `mod hints` that could give you a-la-Rust error messages: _"You passed value X but a Symbol is required. Did you try passing X in the following format :Y? Remember Symbols are 10chars and your is Z chars."_